packages/core/src/node/messaging/messaging-contribution.ts

```diff
@@ -75,15 +74,16 @@ export class MessagingContribution implements BackendApplicationContribution, Me
 
     listen(spec: string, callback: (params: MessagingService.PathParams, connection: MessageConnection) => void): void {
         this.wsChannel(spec, (params, channel) => {
-            const connection = createWebSocketConnection(channel, new ConsoleLogger());
+            const connection = Channel.createMessageConnection(channel, new ConsoleLogger());
             callback(params, connection);
         });
     }
 
-    forward(spec: string, callback: (params: MessagingService.PathParams, connection: IConnection) => void): void {
+    forward(spec: string, callback: (params: MessagingService.PathParams, connection: TheiaMessageConnection) => void): void {
```

Comment:
Is this even ever used? I didn't find any usages.

Reply:
Not anymore but despite the amount of changes in this PR I tried to only touch what was broken/bad.

This method is not bothering anyone so I'd keep it for now :)